src/main/java/net/snowflake/client/config/SFClientConfigParser.java

```diff
@@ -25,41 +29,77 @@ public class SFClientConfigParser {
    *     properties
    * @return SFClientConfig
    */
   public static SFClientConfig loadSFClientConfig(String configFilePath) throws IOException {
+    if (configFilePath != null) {
+      logger.info("Attempting to enable easy logging with file path {}", configFilePath);
+    }
     String derivedConfigFilePath = null;
     if (configFilePath != null && !configFilePath.isEmpty()) {
       // 1. Try to read the file at  configFilePath.
       derivedConfigFilePath = configFilePath;
+      logger.info(
+          "Using client configuration path from a connection string: {}", derivedConfigFilePath);
     } else if (System.getenv().containsKey(SF_CLIENT_CONFIG_ENV_NAME)) {
       // 2. If SF_CLIENT_CONFIG_ENV_NAME is set, read from env.
       derivedConfigFilePath = systemGetEnv(SF_CLIENT_CONFIG_ENV_NAME);
+      logger.info(
+          "Using client configuration path from an environment variable: {}",
+          derivedConfigFilePath);
     } else {
       // 3. Read SF_CLIENT_CONFIG_FILE_NAME from where jdbc jar is loaded.
       String driverLocation =
           Paths.get(getConfigFilePathFromJDBCJarLocation(), SF_CLIENT_CONFIG_FILE_NAME).toString();
       if (Files.exists(Paths.get(driverLocation))) {
         derivedConfigFilePath = driverLocation;
+        logger.info(
+            "Using client configuration path from jar directory: {}", derivedConfigFilePath);
       } else {
         // 4. Read SF_CLIENT_CONFIG_FILE_NAME if it is present in user home directory.
-        String userHomeFilePath =
-            Paths.get(systemGetProperty("user.home"), SF_CLIENT_CONFIG_FILE_NAME).toString();
-        if (Files.exists(Paths.get(userHomeFilePath))) {
-          derivedConfigFilePath = userHomeFilePath;
+        String homePath = systemGetProperty("user.home");
+        if (homePath != null) {
+          String userHomeFilePath = Paths.get(homePath, SF_CLIENT_CONFIG_FILE_NAME).toString();
+          if (Files.exists(Paths.get(userHomeFilePath))) {
+            derivedConfigFilePath = userHomeFilePath;
+            logger.info(
+                "Using client configuration path from home directory: {}", derivedConfigFilePath);
+          }
         }
       }
     }
     if (derivedConfigFilePath != null) {
       try {
+        if (Constants.getOS() != Constants.OS.WINDOWS) {
+          // Check permissions of config file
+          Set<PosixFilePermission> folderPermissions =
+              Files.getPosixFilePermissions(Paths.get(derivedConfigFilePath));
+          if (folderPermissions.contains(PosixFilePermission.GROUP_WRITE)
+              || folderPermissions.contains(PosixFilePermission.OTHERS_WRITE)) {
+            String error =
+                String.format(
+                    "Error due to other users having permission to modify the config file: %s",
+                    derivedConfigFilePath);
+            throw new IOException(error);
```

Comment:
because of this exception we should consider the issue as a BCR

Reply:
Are you sure? IOException was thrown before.

Reply:
Here we have additional check that when GROUP or OTHERS may write than the exception is thrown - we didn't throw such exception before

Reply:
@sfc-gh-ext-simba-nl please change this place to only log on warn level and add a TODO comment with jira SNOW-1503722 to change it in the future